### PR TITLE
Set author to UNDEFINED if no author given

### PR DESCRIPTION
**Pull request by tuxlife**

### CODE REVIEW

File: scm_source/api.py [hunk 1]
```diff
@@ -15,6 +15,8 @@ def get_scm_source_data(author: str):
     status = get_output(['git', 'status', '--porcelain'])
     if status:
         rev = '{} (locally modified)'.format(rev)
+    if author is None:
+        author = 'UNDEFINED'
     data = {'url': 'git:{}'.format(url), 'revision': rev, 'author': author, 'status': status}
```
Comment (hjacobs) [inline scm_source/api.py:20]: IMHO much better would be to use:

``` python
'author': author or 'unknown'
```

Comment (tuxlife) [inline scm_source/api.py:20]: I think, this is not better.

the main problem is the `null` in the json. Not a empty string.

``` python
>>> author = ''
>>> author or 'unknown'
'unknown'
```

Comment (hjacobs) [inline scm_source/api.py:20]: I think the empty string has the same meaning as "null" here (i.e. we don't know the author) -> replacing it with "unknown" is fine IMO

**Comment by hjacobs** [comment]
This PR is no longer needed as @tuxlife changed it in master :see_no_evil:

**Pull request closed without merging**